plugin-server/src/main/ingestion-queues/kafka-queue.ts

```diff
@@ -166,20 +167,18 @@ type EachBatchFunction = (messages: Message[], queue: IngestionConsumer) => Prom
 
 export class IngestionConsumer {
     public pluginsServer: Hub
-    public consumerReady: boolean
     public topic: string
     public consumerGroupId: string
     public eachBatch: EachBatchFunction
     public consumer?: BatchConsumer
+    public eventsProcessor: EventsProcessor
 
     constructor(pluginsServer: Hub, topic: string, consumerGroupId: string, batchHandler: EachBatchFunction) {
         this.pluginsServer = pluginsServer
         this.topic = topic
         this.consumerGroupId = consumerGroupId
-
-        this.consumerReady = false
-
         this.eachBatch = batchHandler
+        this.eventsProcessor = new EventsProcessor(pluginsServer)
```

Comment:
This is the fix from before. Moving the eventsProcessor here ensures it is only created once per consumer rather than before where it ended up being once per event